tools/diagnostics/nsdiff

```diff
@@ -0,0 +1,875 @@
+#!/usr/bin/env python
+
+# README:
+#
+# This script assumes:
+# --------------------
+#
+# 1. The entirety of a node's state is bundled as a zip or tar file
+# 2. The zip/tar file has the name <node_name>.zip or <node_name>.tar.gz where
+#    <node_name> is the name of the node.
+# 3. Both zip/tar files to be compared are on the filesystem where nsdiff is
+#    installed. In other words, nsdiff does not attempt to connect or collect
+#    state data from remote nodes.
+# 4. nscapture is/was used to capture state passed as state1 and state2
+#
+# Installation:
+# -------------
+#
+# If this script is used on a server or workstation that has indy-node installed
+# and indy-node is known to be working, you do not need to install anything.
+# Otherwise, the following information will be helpful.
+#
+# This script depends on leveldb/rocksdb python bindings which in turn rely on
+# the leveldb C runtime libs and headers. You must install leveldb/rocksdb
+# dependencies. See Installation > Prerequisites section below.
+#
+# Until further notice, assume Python 3.5.2 is required and all references to
+# python3 or pip3 imply Python 3.5.2.
+# 
+# Note that anytime pip/pip3 is used to install python packages, the
+# python/python3 environment is changed without the knowledge of package
+# managers (apt, yum, brew, etc.). Side-effects of pip installations and
+# os-controlled updates (i.e. automatic updates) to your system's python have
+# the potential to cause instability/breakage of either your system or
+# development environment. Most modern operating systems depend on Python. If
+# you would prefer not to make changes to your system python/python3, please
+# consider setting up a Python Virtual Environment (highly recommended):
```

Comment:
FROM ALEX
I think in most of the cases this script will be run on a real installation from deb packages (for example in Docker), not with local nodes.